File: app/src/main/java/com/cheatdatabase/data/dao/SystemDao.java
package com.cheatdatabase.data.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.cheatdatabase.data.model.SystemModel;

import java.util.List;

@Dao
public interface SystemDao {

    @Query("SELECT * FROM systems")
    List<SystemModel> getAll();

    @Query("SELECT * FROM systems where _id = :systemId")
    SystemModel getSystemById(int systemId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(SystemModel systemModel);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    List<Long> insertAll(List<SystemModel> systemModels);

    @Delete
    void delete(SystemModel systemModel);

    @Query("DELETE FROM systems")
    void deleteAll();

    @Update
    void update(SystemModel systemModel);

    @Query("DELETE FROM systems where _id = :systemId")
    void deleteBySystemId(int systemId);


}

File: app/src/main/java/com/cheatdatabase/data/dao/SystemDao.kt
package com.cheatdatabase.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.cheatdatabase.data.model.SystemModel

@Dao
interface SystemDao {

    @Query("SELECT * FROM systems")
    fun loadAllSystems(): Array<SystemModel>

    @Query("SELECT * FROM systems")
    fun all(): LiveData<List<SystemModel>>

    @Query("SELECT * FROM systems ORDER BY name")
    suspend fun allAsList(): List<SystemModel?>?

    @Query("SELECT * FROM systems where _id = :systemId")
    fun getSystemById(systemId: Int): LiveData<SystemModel?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(systemModel: SystemModel): Long?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(systemModels: List<SystemModel>): List<Long>

    @Delete
    fun delete(systemModel: SystemModel)

    @Query("DELETE FROM systems")
    fun deleteAll()

    @Update
    fun update(systemModel: SystemModel)

    @Query("DELETE FROM systems where _id = :systemId")
    fun deleteBySystemId(systemId: Int)
}